ruma-identifiers-validation/src/crypto_algorithms.rs

```diff
@@ -67,7 +67,8 @@ impl TryFrom<String> for ServerKeyAlgorithm {
 }
 
 /// An encryption algorithm to be used to encrypt messages sent to a room.
-#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[derive(Clone, Debug, PartialEq)]
+#[cfg_attr(feature = "serde", derive(Deserialize, Serialize))]
 #[non_exhaustive]
 #[serde(from = "String", into = "String")]
```

Comment:
I'm a bit confused that this isn't erroring. Maybe CI for `ruma-identifiers` with `default-feature = false` is broken. These attributes need to be part of the `cfg_attr` as well (like the serde attributes for the other enums).

Reply:
Seems like it was yet another bug introduced when moving the validation logic and enums into `ruma-identifiers-validation`. Fixed in https://github.com/ruma/ruma/commit/585485831b700a86b82ae9fe29dd50e947bb416b.